components/GamePage.jsx

```diff
@@ -1,59 +1,110 @@
 "use client";
 import { useState, useEffect, useContext } from "react";
 
+import * as _ from "lodash";
+
 import { GameContext } from "@/contexts/GameContextProvider";
 
 import createWinningCombinationsArrays from "@/app/utils/createWinningCombinationsArrays";
-import splitArrayIntoChunks from "@/app/utils/splitArrayIntoChunks";
+import checkSimilarityOfArrays from "@/app/utils/checkSimilarityOfArrays";
+import Modal from "./Modal";
 
 export default function GamePage() {
-  const { playerType, setPlayerType, gameType, setGameType } =
+  const { playerType, setPlayerType, gameType, setGameType, setWinner } =
     useContext(GameContext);
 
+  const [crossIndex, setCrossIndex] = useState([]);
+  const [circleIndex, setCircleIndex] = useState([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
   const initialGameBoard = [
     [1, 2, 3],
     [4, 5, 6],
     [7, 8, 9],
   ];
-
   const [gameBoard, setGameBoard] = useState(initialGameBoard);
 
   const handleGameBoardClick = (cell) => {
     const row = Math.floor((cell - 1) / 3);
     const column = (cell - 1) % 3;
 
-    const newGameBoard = [...gameBoard];
+    if (playerType === "X") {
+      setCrossIndex((prevState) => [
+        ...prevState,
+        initialGameBoard[row][column],
+      ]);
+    } else {
+      setCircleIndex((prevState) => [
+        ...prevState,
+        initialGameBoard[row][column],
+      ]);
+    }
 
+    const newGameBoard = [...gameBoard];
     newGameBoard[row][column] = playerType;
-
     setGameBoard(newGameBoard);
-
     setPlayerType((prevPlayerType) => (prevPlayerType === "X" ? "O" : "X"));
   };
 
+  const { rows, columns, leftDiagonal, rightDiagonal } =
+    createWinningCombinationsArrays(initialGameBoard);
```

Comment:
instead of returning separate arrays just take return this 

```js
const algorithm = [
       ..._.chunk(rows, 3),
       ..._.chunk(columns, 3),
       leftDiagonal,
       rightDiagonal,
     ];
```

from `createWinningCombinationsArrays` function